src/collective/collectionfilter/vocabularies.py

```diff
@@ -109,6 +112,16 @@ def translate_portal_type(value, *args, **kwargs):
     return term.title if term else value
 
 
+def _groupby_cache_key(method, self):
+    portal = plone.api.portal.get()
+    # cache key is shared across sites, so path is included
+    site_path = '/'.join(portal.getPhysicalPath())
+    catalog = queryUtility(ICatalogTool)
+    if catalog is None:
+        return site_path
+    return '%s-%s' % (site_path, str(catalog.getCounter()))
```

Comment:
I would recommend to add the current logged users permissions here too ... there might be differences in the groupCriteria when you're permitted to see more results.

Reply:
@petschki thanks for looking at this and your feedback.

I have looked over the catalog and the code that creates the group by criteria and I am sure that the list of items does not change based on user permissions.

The `groupby` function is calling the `schema()` method of `portal_catalog` which is protected by [`AccessControl.Permissions.search_zcatalog`](https://github.com/zopefoundation/Products.ZCatalog/blob/5.x/src/Products/ZCatalog/ZCatalog.py#L579). As far as I can see, individual indexes aren't permission aware, either you can list the indexes or you can't.

So therefore I would argue that adding user permissions to the cache key is adding unnecessary overhead for no value.

That said, it is a somewhat trivial change to add user permissions to the key if you still feel it is necessary?